rdt-open-reader/src/covidIndonesia/java/io/ona/rdt/widget/OneScanCovidRDTBarcodeFactory.java

```diff
@@ -39,4 +40,9 @@ protected String getBarcodeValsAsCSV(Intent data) {
     protected String[] splitCSV(String barcodeCSV) {
         return barcodeCSV.split(",");
     }
-}
+
+    @Override
+    protected void sensorTriggered() {
+        ((RDTJsonFormFragmentPresenter) widgetArgs.getFormFragment().getPresenter()).moveToNextStep("step2");
```

Comment:
The step to navigate to should be fetched from the `StepStateConfig`.

Also, confirm that step2 has the correct message as specified in the issue.

Reply:
fixed.